packages/components/src/components/link/_link.scss

```diff
@@ -95,6 +96,12 @@
   .#{$prefix}--link--lg {
     @include type-style('body-short-02');
   }
+
+  .#{$prefix}--link__icon {
```

Comment:
Would be great to get the following snippet to "Just work":

```jsx
<Link href="...">Text <SomeIcon /></Link>
```

I'm assuming the big challenge is adding in the correct spacing for the icon (in this case `$spacing-03`) so we need a selector to target it and then need to provide that to the component itself, correct?

Just thinking out loud, this may be what we have to end up going with now, but would be interesting to explore either "gap" or custom property based solutions in the future. Gap would be interesting in the context of grid (and flex when it gets it) since I think it only is applied when there are more than one items. `display: inline-grid` plus `grid-column-gap` and `grid-auto-flow: columns` may be a nice trick in the future for these.

Reply:
Interesting idea. We couldn't directly prevent the misuse of inline links with icons then though, could we?

Reply:
Maybe we add in a class for the "inline" link, I guess? Then scope the CSS rules outside of that.

Definitely thinking for later on, right now this seems like a solid step given what our API and constraints are 👍